src/basic/macro.h

```diff
@@ -405,10 +405,20 @@ static inline int __coverity_check_and_return__(int condition) {
                 func(p);                                        \
         }
 
+/* When func() returns the appropriate type */
```

Comment:
comment should say *NULL cast to the appropriate type*.

i.e. the NULL matters

Reply:
Actually it's not NULL — in some cases it's e.g. -1. If it was always NULL, I wouldn't need the second variant of DEFINE_TRIVIAL_CLEANUP_FUNC. I'll try to improve the comment.